### DISCUSSION TITLE
Is the generated asset file supposed to be added to version control?

### DISCUSSION THREAD

**Discussion by smac89** [webdeveric/webpack-assets-manifest#228] [Q&A] [answered]
I'm interested in using this plugin, but I wanted to ask how we are supposed to manage the generated manifest file. Is it supposed to be tracked using something like git?

How does one use it between regular development (webpack-dev-server) vs when actually building the application for deployment?

I've read another [blog](https://slack.engineering/keep-webpack-fast-a-field-guide-for-better-build-performance/) which mentions the manifest file, but no-one seems to talk about proper usage

**Comment by webdeveric** [comment, marked as answer]
@smac89 I don't commit the manifest to git. I usually have a build and deploy step where the build artifact is generated then deployed somewhere.

The usage of this plugin during development is entirely up to you and how the assets manifest will be used in your codebase.

**Reply by smac89** [reply]
Ok can you please explain that.

What build artifacts are generated and how does the manifest file get used in either the build or deploy stage?

**Reply by webdeveric** [reply]
Build artifacts are just the output from webpack, which includes the `assets-manifest.json` file that this plugin generates.

How you use the assets manifest is your choice. Normally it is used by your app so you can lookup a hashed filename when you only know the original filename.

**Reply by smac89** [reply]
I see. So the manifest is used as a lookup table for a server?

What kind of server-side application do you use which makes use of this manifest?

**Reply by webdeveric** [reply]
Essentially, yes, that is one use case.
I've personally used the assets manifest on a Laravel app, WordPress site, and a variety of Node apps.

https://github.com/webdeveric/next-assets-manifest
https://github.com/webdeveric/assemble-helper-assets-manifest

This plugin is currently being used by [rails/webpacker](https://github.com/rails/webpacker), if you prefer coding in `ruby`.

**Reply by smac89** [reply]
I was able to find more examples of using the manifest file here:
https://developers.google.com/web/fundamentals/performance/webpack/use-long-term-caching
